connectors/utils.py

```diff
@@ -432,18 +432,22 @@ def wrapper(func):
             processed_skipped_exceptions = skipped_exceptions
 
         if inspect.isasyncgenfunction(func):
-            return retryable_async_generator(
+            return retryable_asyncgen_function(
+                func, retries, interval, strategy, processed_skipped_exceptions
+            )
+        elif inspect.iscoroutinefunction(func):
+            return retryable_coroutine_function(
                 func, retries, interval, strategy, processed_skipped_exceptions
             )
         else:
```

Comment:
I'm not sure this covers all. We'd better be more defensive.

```suggestion
        elif inspect.isfunction(func):
            return retryable_sync_function(
                func, retries, interval, strategy, processed_skipped_exceptions
            )
        else:
            raise NotSupportedObjectError(f"Not supported {func.__class__}")
```